### PR TITLE
(BA-1077) baseapp modularization

### PR DESCRIPTION
**Pull request by anicioalexandre**
* authentication package creation - `v1.0.0`
  - Creates Authentication package.
  - Creates `access`, `mfa` and `user` modules alongside its services api's and types.
  - Migrate authentication methods and tests from the `core` package.
 
* config package update - `v2.1.0`
  - Update ESlint `ignorePatterns`, `extends` and `plugins`.
  - Update plugin`s dependencies.
 
* core package update - `v2.7.0`
  - The `createAxiosInstance` function got parameter changes and small refactors.
  - The `react-query` package was updated to `@tanstack/react-query` alongside with adaptions required for this update.

* design-system-mui package update - `v1.4.0`
  - Changes `ButtonWithLoading` props in order to remove `form`.
  - Adds a generic type to `CheckboxField`.
  - Creates `ThemeProvider` component.

* provider package creation - `v1.0.0`
  - Creates Provider package.
  - Creates `ReactQueryProvider` component.

* test package creation - `v1.0.0`
  - Creates Test package.
  - Exports React Testing Library features.
  - Creates functions, mocks and test configurations.

* tsconfig package update - `v1.1.0`
  - Updates `nextjs.json` configuration.

* utils package creation - `v1.0.0`
  - Creates Provider package.
  - Creates constants, functions, hooks and types to be reused between apps and packages.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,10 +40,15 @@ export default function Docs() {
 
 ## Apps and Packages
 
-- `docs`: a app to document some packages's features
-- `core`: core of utilities like `auth hooks`, `permisisons system` and `util functions`
-- `config`: reusable configurations for `eslint`, `prettier` and `jest`
-- `tsconfig`: reusable typescript configs
+- `docs`: an app to document some packages's features
+- `authentication`: includesauthentication modules such as `login`, `signup`, `reset password`, `multifactor authentication` and more.
```
Comment (deboracosilveira) [inline README.md]: includes authentication

File: packages/test/utils/mocks.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+import MockAdapter from 'axios-mock-adapter'
+import Cookies from 'js-cookie'
+
+jest.mock('js-cookie')
+
+export type CookiesGetByNameFn = (name: string) => string
+
+export const cookiesMock = jest.mocked(Cookies, true)
+// @ts-ignore TODO: investigate AxiosRequestHeaders error
```
Comment (deboracosilveira) [inline packages/test/utils/mocks.ts]: like RR said on the other PR, lets add TODOS to some story?

**Review by deboracosilveira — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (anicioalexandre) [reply README.md]: thanks!

File: packages/test/utils/mocks.ts (hunk 2 above)
Comment (anicioalexandre) [reply packages/test/utils/mocks.ts]: good catch, created this one for all TS errors https://app.approvd.io/projects/BA/stories/25108

**Review by ficast — APPROVED** [review]

**Pull request merged**